oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -171,6 +172,9 @@ public static GoogleCredentials fromStream(
     if (SERVICE_ACCOUNT_FILE_TYPE.equals(fileType)) {
       return ServiceAccountCredentials.fromJson(fileContents, transportFactory);
     }
+    if (GDCH_SERVICE_ACCOUNT_FILE_TYPE.equals(fileType)) {
+      return GdchCredentials.fromJson(fileContents, transportFactory);
```

Comment:
have a constructor without transport, and create a correct transport internally. Because there is no reason to pass the one we will never use

Reply:
done